framework-plugins/lisk-framework-chain-connector-plugin/src/chain_connector_plugin.ts

```diff
@@ -355,13 +380,66 @@ export class ChainConnectorPlugin extends BasePlugin<ChainConnectorPluginConfig>
 			this._ccuFrequency.liveness >= newBlockHeader.height - this._lastCertifiedHeight
 		) {
 			// TODO: _createCCU needs to be implemented which will create and send the CCU transaction
+			const listOfCCMs = this._getListOfCCMs(
+				this._chainConnectorState.crossChainMessages,
+				newBlockHeader,
+			);
+			// eslint-disable-next-line no-console
+			console.log(listOfCCMs);
 			await this._createCCU();
 			// if the transaction is successfully submitted then update the last certfied height and do the cleanup
 			// TODO: also check if the state is growing, delete everything from the inMemory state if it goes beyond last 3 rounds
-			await this._cleanup();
+			this._cleanup();
 		}
 	}
 
+	private _getListOfCCMs(
+		crossChainMessages: CcmWithHeight[],
+		newBlockHeader: BlockHeader,
+	): CcmWithHeight[][] {
+		const filteredCCMs = crossChainMessages.filter(
+			cmmWithHeight => cmmWithHeight.height >= newBlockHeader.aggregateCommit.height,
+		);
+
+		const listOfCCMs: CcmWithHeight[][] = [];
+
+		const getListWithNewIndex = (
+			ccms: CcmWithHeight[],
+			startIndex: number,
+		): [list: CcmWithHeight[], newIndex: number] => {
+			const list: CcmWithHeight[] = [];
+			let totalSize = 0;
+			let newIndex = startIndex;
+
+			for (let i = startIndex; i < ccms.length; i += 1) {
+				const ccmWithHeight = filteredCCMs[i];
+				const ccmBytes = codec.encode(ccmSchema, ccmWithHeight.ccm);
+				const size = ccmBytes.length;
+				totalSize += size;
+				if (totalSize > CCU_TOTAL_CCM_SIZE) {
+					break;
+				}
+
+				newIndex = i;
+				list.push(ccmWithHeight);
+			}
+
+			return [list, newIndex + 1];
+		};
+		const getListOfList = (inputList: CcmWithHeight[][], startIndex: number) => {
+			const [list, newIndex] = getListWithNewIndex(filteredCCMs, startIndex);
+			inputList.push(list);
+
+			if (newIndex < filteredCCMs.length) {
```

Comment:
based on the above comment, you can do below?
```suggestion
			if (newIndex < filteredCCMs.length - 1) {
```